docs/development/workflow/development_workflow.rst

```diff
@@ -595,22 +584,26 @@ made conflict with changes that someone else made — than anywhere else. Ask fo
 help if you need it. Instructions are available on how to
 `resolve merge conflicts after a Git rebase <https://help.github.com/en/articles/resolving-merge-conflicts-after-a-git-rebase>`_.
 
+.. _squash-if-necessary:
+
+Squash if necessary
+*******************
+
+The :ref:`when-to-squash-commits` section describes when to squash commits.
+
+Squashing commits changes the metadata/commit hash, and also removes separate visibility
+of individual commits; a new commit and commit message will only contain a textual list
+of the earlier commits.
+
 .. _howto_squash:
 
 How to squash
-*************
-
-Typically we ask to *squash* when there was a fair amount of trial
-and error, but the final patch remains quite small, or when files were added
-and removed (especially binary files or files that should not remain in the
-repository) or if the number of commits in the history is disproportionate
-compared to the work being carried out (for example 30 commits gradually
-refining a final 10-line change).  Conceptually this is equivalent to
-exporting the final diff from a feature branch, then starting a new branch and
-applying only that patch.
-
-Many of us find that is it actually easiest to squash using rebase. In particular,
-you can rebase and squash within the existing branch using::
+=============
+
+In many cases squashing can be done using the :ref:`github-squash-and-merge` button on
+the GitHub pull request page. If this is not possible we typically squash using an
+interactive git rebase. In particular, you can rebase and squash within the existing
```

Comment:
This would be a good change, but I'm not entirely sure I got the syntax right.
```suggestion
the GitHub pull request page. If this is not possible we typically squash using
```git rebase --interactive`` <https://git-scm.com/docs/git-rebase#_interactive_mode>`_. In particular, you can rebase and squash within the existing
```